lib/ui/common/roundedButton.dart

```diff
@@ -2,125 +2,152 @@ import 'package:flutter/material.dart';
-  final TextStyle buttonTextStyle = const TextStyle(fontSize: 17, fontWeight: FontWeight.bold, color: Color(0xffffffff));
 
-  final double iconEdgePadding = 5;
-  final double size = 56.0;
-  final double buttonIconSize = null;
+    );
+  }
 
-  const _DefaultStyle();
-}
+  factory RoundedButtonStyle.actionSheetLargeRoundedButton(){
+    return RoundedButtonStyle.largeRoundedButtonStyle().copyWith(
+      backgroundColor: Color(0xFFe9eaf2),
+      buttonTextStyle: TextStyle(fontSize: 15, fontWeight: FontWeight.w500, color: Color(0xFF4c4e6e)),
+    );
+  }
 
-class RoundedButton {
-  static build(
-      {RoundedButtonStyle style = const _DefaultStyle(),
-        @required  BuildContext context,
-        String title, IconData icon,
-        Function onTap}) {
-    return _buildRoundedButton(style, context , title: title, icon: icon, onTap: onTap);
+  RoundedButtonStyle copyWith({Color iconButtonColor, Color backgroundColor,
+    BoxShape buttonShape, BorderRadius borderRadius, EdgeInsets edgePadding,
+    TextStyle buttonTextStyle, double height, double width, double iconEdgePadding,
+    double buttonIconSize}) {
+
+    return RoundedButtonStyle (
+        iconButtonColor: iconButtonColor ?? this.iconButtonColor,
+        backgroundColor: backgroundColor ?? this.backgroundColor,
+        buttonShape: buttonShape ?? this.buttonShape,
+        borderRadius: borderRadius ?? this.borderRadius,
+        edgePadding: edgePadding ?? this.edgePadding,
+        buttonTextStyle: buttonTextStyle ?? this.buttonTextStyle,
+        height: height ?? this.height,
+        iconEdgePadding: iconEdgePadding ?? this.iconEdgePadding,
+        width: width ?? this.width,
+        buttonIconSize: buttonIconSize ?? this.buttonIconSize
+    );
   }
+}
 
-  static Widget _buildRoundedButton(RoundedButtonStyle buttonStyle, BuildContext context, {String title, IconData icon,
-    Function onTap}) {
 
-    RoundedButtonViewModel viewModel = buildButtonViewModel(context, title: title, icon: icon, onTap: onTap);
+class RoundedButton  extends StatelessWidget {
+  final RoundedButtonViewModel _viewModel;
+  final RoundedButtonStyle _style;
+
+  const RoundedButton({Key key, RoundedButtonViewModel viewModel, RoundedButtonStyle style })
+      : this._viewModel = viewModel,
+        this._style = style,
+        super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return GestureDetector(
+      onTap: () => _viewModel.onTap(),
+      child: Column(
+        mainAxisAlignment: MainAxisAlignment.end,
+        children: <Widget>[
+          Padding(
+            padding: _style.edgePadding,
+            child: Container(
+              alignment: Alignment.center,
+              height: _style.height,
+              width: _style.width,
+              decoration: BoxDecoration(
+                  color: _viewModel.backgroundColor != null ? _viewModel.backgroundColor : _style.backgroundColor,
```

Comment:
why viewmodel involved here? that's not the place for colours . it can have things like "isSelected" and then the style has selected and non selected styles. But it shouldn't drive the style like this.